internal/jimm/role/role_test.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2024 Canonical.
+package role
+
+import (
+	"context"
+	"sort"
+	"strconv"
+	"testing"
+	"time"
+
+	cofga "github.com/canonical/ofga"
+	qt "github.com/frankban/quicktest"
+	"github.com/frankban/quicktest/qtsuite"
+
+	"github.com/canonical/jimm/v3/internal/common/pagination"
+	"github.com/canonical/jimm/v3/internal/db"
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/openfga"
+	ofganames "github.com/canonical/jimm/v3/internal/openfga/names"
+	jimmtest "github.com/canonical/jimm/v3/internal/testutils/jimmtest"
+)
+
+type roleManagerSuite struct {
+	manager    *roleManager
+	user       *openfga.User
+	db         *db.Database
+	ofgaClient *openfga.OFGAClient
+}
+
+func (s *roleManagerSuite) Init(c *qt.C) {
+	// Setup DB
+	db := &db.Database{
+		DB: jimmtest.PostgresDB(c, time.Now),
+	}
+	err := db.Migrate(context.Background(), false)
+	c.Assert(err, qt.IsNil)
+
+	s.db = db
+
+	// Setup OFGA
+	ofgaClient, _, _, err := jimmtest.SetupTestOFGAClient(c.Name())
+	c.Assert(err, qt.IsNil)
+
+	s.ofgaClient = ofgaClient
+
+	s.manager, err = NewRoleManager(db, ofgaClient)
+	c.Assert(err, qt.IsNil)
+
+	// Create test identity
+	i, err := dbmodel.NewIdentity("alice")
+	c.Assert(err, qt.IsNil)
+	s.user = openfga.NewUser(i, ofgaClient)
+}
+
+func (s *roleManagerSuite) TestAddRole(c *qt.C) {
+	c.Parallel()
```

Comment:
what's this?

Reply:
Allows the tests to run in parallel :) bit of a premature optimisation.

Reply:
if it works just like that is cool!

Reply:
Well I never checked if Fran plugged it in but I'm presuming it's calling underlying TB parallel? Can check?

Reply:
My question is: can we just use it without worrying about test race conditions?

Reply:
I believe each test but need to trawl down a bit, I just tested this via creating new tests against the same object. Fran's quicktest doesn't have the before/after suite/test methods, just an init... Which I guess is another discussion we need to have.

Reply:
Or do they share the same db instance, so should we be careful?

Reply:
In any case we create db by name, so it's safe - I don't believe they're separate instances if that's what you mean?

Reply:
If the `init` is called again each time before running the tests, then they don't share the same db. 
But if they use the same `s.db` they share same db and table. 
Right? 
If the latter running tests in parallel is not trivial

Reply:
thank you!

Reply:
so the c.parallel should be kind of a "free lunch"

Reply:
Tests from different packages are already run in parallel from what I understand btw. For more info
https://stackoverflow.com/a/72354980